src/driver_gpio.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 <Admobilize>
+ * MATRIX Labs  [http://creator.matrix.one]
+ * This file is part of MATRIX Creator MALOS
+ *
+ * MATRIX Creator MALOS is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <iostream>
+
+#include "./driver_gpio.h"
+#include "./src/driver.pb.h"
+
+namespace matrix_malos {
+
+const bool kGpioDriverDebugEnabled = false;
+
+bool GpioDriver::ProcessConfig(const DriverConfig& config) {
+  GpioParams gpio_config(config.gpio());
+  int16_t pin = (int16_t)gpio_config.pin();
```

Comment:
Can you try without the casts? (int16_t) Do you get an overflow warning without them?

In case you need them, the C++ way to do it is static_cast<int16_t>(pio_config.pin())

Also, *pin* and *mode* can be declared as *const* as they're not modified in the function.